cluster/gce/gci/configure-helper.sh

```diff
@@ -0,0 +1,684 @@
+    client-key-data: ${KUBELET_KEY}
+clusters:
+- name: local
+  cluster:
+    certificate-authority-data: ${KUBELET_CA_CERT}
+contexts:
+- context:
+    cluster: local
+    user: kubelet
+  name: service-account-context
+current-context: service-account-context
+EOF
+}
+
+# Uses KUBELET_CA_CERT (falling back to CA_CERT), KUBELET_CERT, and KUBELET_KEY
+# to generate a kubeconfig file for the kubelet to securely connect to the apiserver.
+function create-master-kubelet-auth {
+  # Only configure the kubelet on the master if the required variables are
+  # set in the environment.
+  if [[ -n "${KUBELET_APISERVER:-}" && -n "${KUBELET_CERT:-}" && -n "${KUBELET_KEY:-}" ]]; then
+    create-kubelet-kubeconfig
+  fi
+}
+
+function create-kubeproxy-kubeconfig {
+  echo "Creating kube-proxy kubeconfig file"
+  cat > /var/lib/kube-proxy/kubeconfig << EOF
+apiVersion: v1
+kind: Config
+users:
+- name: kube-proxy
+  user:
+    token: ${KUBE_PROXY_TOKEN}
+clusters:
+- name: local
+  cluster:
+    certificate-authority-data: ${CA_CERT}
+contexts:
+- context:
+    cluster: local
+    user: kube-proxy
+  name: service-account-context
+current-context: service-account-context
+EOF
+}
+
+function assemble-docker-flags {
+  local docker_opts="-p /var/run/docker.pid --bridge=cbr0 --iptables=false --ip-masq=false"
+  if [[ "${TEST_CLUSTER:-}" == "true" ]]; then
+    docker_opts="${docker_opts} --debug"
+  fi
+  echo "DOCKER_OPTS=\"${docker_opts} ${EXTRA_DOCKER_OPTS:-}\"" > /etc/default/docker
+}
+
+# Loads kube-system docker images. It is better to do it before starting kubelet,
+# as kubelet will restart docker daemon, which may interfere with loading images.
+function load-docker-images {
+  echo "Start loading kube-system docker images"
+  local -r img_dir="${KUBE_HOME}/kube-docker-files"
+  if [[ "${KUBERNETES_MASTER:-}" == "true" ]]; then
+    echo "Try to load docker image file kube-apiserver.tar"
+    timeout 30 docker load -i "${img_dir}/kube-apiserver.tar"
+    echo "Try to load docker image file kube-controller-manager.tar"
+    timeout 30 docker load -i "${img_dir}/kube-controller-manager.tar"
+    echo "Try to load docker image file kube-scheduler.tar"
+    timeout 30 docker load -i "${img_dir}/kube-scheduler.tar"
+  else
+    echo "Try to load docker image file kube-proxy.tar"
+    timeout 30 docker load -i "${img_dir}/kube-proxy.tar"
+  fi
+}
+
+# A kubelet systemd service is built in GCI image, but by default it is not started
+# when an instance is up. To start kubelet, the command line flags should be written
+# to /etc/default/kubelet in the format "KUBELET_OPTS=<flags>", and then start kubelet
+# using systemctl. This function assembles the command line and start the kubelet
+# systemd service.
+function start-kubelet {
+  echo "Start kubelet"
+  local flags="--v=2"
+  if [[ -n "${KUBELET_TEST_LOG_LEVEL:-}" ]]; then
+    flags="${KUBELET_TEST_LOG_LEVEL}"
+  fi
+  if [[ -n "${KUBELET_PORT:-}" ]]; then
+    flags="${flags} --port=${KUBELET_PORT}"
```

Comment:
This might be easier to read as `flags+=("--port=${KUBELET_PORT}")` like @zmerlynn did in #25437.